### PR TITLE
A lower bound on python version for noarch python is now required

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Review by ocefpaf — APPROVED** [review]

**Review by mariusvniekerk — APPROVED** [review]
This is a good move

**Comment by isuruf** [comment]
@beckermr, do you know why codecov is failing?

**Review by chrisburr — APPROVED** [review]

File: tests/test_lint_recipe.py [hunk 1]
```diff
@@ -582,9 +582,8 @@ def test_bad_requirements_order(self):
 
     def test_noarch_python_bound(self):
         expected_message = (
-            "noarch: python recipes are recommended to have a lower bound "
-            "on the python version. This recommendation will become a "
-            "requirement in the future."
+            "noarch: python recipes are required to have a lower bound "
+            "on the python version."
```
Comment (chrisburr) [inline tests/test_lint_recipe.py:586]: How about explaining the solution a little more explicitly?
```suggestion
            "noarch: python recipes are required to have a lower bound "
            "on the python version. Typically this means putting "
            "`python >=3.6` in **both** `host` and `run` but you should check "
            "upstream for the package's Python compatibility."
```

**Comment by beckermr** [comment]
Sorry I missed this @isuruf! I have no idea why the changes in the coveralls stuff were needed. :/

**Comment by CJ-Wright** [comment]
Do we need a mini migrator to go with this?

**Comment by ocefpaf** [comment]
> Do we need a mini migrator to go with this?

That wouldn't hurt but we may have a small corner case for those feedstocks that are still building Python 2.7 using our workaround. I'm not sure how many there are or if they are still needed. (This is not a no, just a thought on a small thing that we may break if we do a migration b/c the default will be `>=3`).

**Pull request merged**